tilequeue/query.py

```diff
@@ -157,16 +157,17 @@ def __init__(self, conn_info, layer_data, io_pool, n_conn):
 
         self.dbnames = self.conn_info.pop('dbnames')
         self.dbnames_query_index = 0
-        self.sql_conn_pool = DBAffinityConnectionsNoLimit(
-            self.dbnames, n_conn, self.conn_info)
+        self.sql_conn_pool = DatabaseCycleConnectionPool(
+            n_conn, 100, self.dbnames, self.conn_info)
```

Comment:
I have a hard-coded 100 connection maximum here but it should probably be a configurable thing. The only time this gets checked is when we're requesting [a new connection out of the pool](https://github.com/psycopg/psycopg2/blob/master/lib/pool.py#L90-L91). The connection pools will only maintain `n_conn` connections per `dbname` otherwise they'll get closed whenever the connection is returned to the pool.